jnigen/lib/src/tools/maven_tools.dart

```diff
@@ -126,17 +153,29 @@ ${depDecls.join("\n")}
   }
 }
 
-/// Maven dependency with group ID, artifact ID, and version.
 class MavenDependency {
   MavenDependency(this.groupID, this.artifactID, this.version,
       {this.otherTags = const {}});
   factory MavenDependency.fromString(String fullName) {
     final components = fullName.split(':');
-    if (components.length != 3) {
+    if (components.length < 3) {
       throw ArgumentError('invalid name for maven dependency: $fullName');
     }
     return MavenDependency(components[0], components[1], components[2]);
   }
   String groupID, artifactID, version;
   Map<String, String> otherTags;
 }
+
+class MavenRepository {
+  MavenRepository(this.id, this.name, this.url, {this.otherTags = const {}});
+  factory MavenRepository.fromString(String fullName) {
+    final components = fullName.split('|');
```

Comment:
This is an undocumented syntax to use `|` as separator?

In case of maven artifacts, the `:` separator between group, artifact and version is universally used convention.

I'd rather read this from YAML alone. (If there's no need to specify it from CLI).

```yaml
repos:
  - id: google
    name: Google Maven Repo
    url: https://maven.google.com
```